src/button-group/item-element.tsx

```diff
@@ -0,0 +1,131 @@
+import IconButtonItem from './icon-button-item.js';
+import MenuDropdownItem from './menu-dropdown-item.js';
+import Tooltip from '../internal/components/tooltip/index.js';
+import styles from './styles.css.js';
+import testUtilStyles from './test-classes/styles.css.js';
+import LiveRegion from '../internal/components/live-region/index.js';
+import clsx from 'clsx';
+
+const ItemElement = forwardRef(
+  (
+    {
+      item,
+      onItemClick,
+    }: {
+      item: ButtonGroupProps.Item;
+      onItemClick?: (event: CustomEvent) => void;
+    },
+    ref: React.Ref<ButtonProps.Ref>
+  ) => {
+    const buttonRef = useRef<HTMLDivElement>(null);
+    const [showTooltip, setShowTooltip] = useState(false);
+    const [showFeedback, setShowFeedback] = useState(false);
+    const popoverFeedback = 'popoverFeedback' in item && item.popoverFeedback;
+    const isIconButton = item.type === 'icon-button';
+    const isMenuDropdown = item.type === 'menu-dropdown';
+
+    const onClickHandler = (event: CustomEvent<ClickDetail>) => {
+      if (popoverFeedback) {
+        setShowFeedback(true);
+        setShowTooltip(true);
+      } else {
+        setShowTooltip(false);
+      }
+
+      fireCancelableEvent(onItemClick, { id: item.id }, event);
+    };
+
+    const onShowTooltip = () => {
+      if (!showTooltip) {
+        setShowFeedback(false);
+        setShowTooltip(true);
+      }
+    };
+
+    const onHideTooltip = () => {
+      if (!showFeedback) {
+        setShowTooltip(false);
+      }
+    };
+
+    useEffect(() => {
+      if (!showTooltip) {
+        return;
+      }
+
+      const close = () => {
+        setShowTooltip(false);
+        setShowFeedback(false);
+      };
+
+      const currentRef = buttonRef.current;
+      const handlePointerDownEvent = (event: PointerEvent) => {
+        if (event.target && currentRef && currentRef.contains(event.target as HTMLElement)) {
+          return;
+        }
+
+        close();
+      };
+
+      const handleKeyDownEvent = (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+          close();
+        }
+      };
+
+      const handleTooltipToogleEvent = (event: CustomEvent) => {
+        if (event.detail.trackKey !== item.id && event.detail.open) {
+          close();
+        }
+      };
+
+      window.addEventListener('pointerdown', handlePointerDownEvent);
+      window.addEventListener('keydown', handleKeyDownEvent);
+      window.addEventListener('btngroup-tooltip:toggle', handleTooltipToogleEvent as any);
+
+      return () => {
+        window.removeEventListener('pointerdown', handlePointerDownEvent);
+        window.removeEventListener('keydown', handleKeyDownEvent);
+        window.removeEventListener('btngroup-tooltip:toggle', handleTooltipToogleEvent as any);
+      };
+    }, [item.id, showTooltip]);
+
+    useEffect(() => {
+      window.dispatchEvent(
+        new CustomEvent('btngroup-tooltip:toggle', { detail: { open: showTooltip, trackKey: item.id } })
```

Comment:
The custom event inside an effect logic is hard to follow and test against. Is it possible to make all checks in the click handler itself?

Instead of having tooltip show and tooltip feedback states here you can lift them to the button group level so that there is only one state for the entire component as there is only one tooltip that can be shown at a time. The state can be like tooltipItemId (can be null which means no tooltip or an item ID which defines for which item to show the tooltip) and tooltipFeedback (a boolean, when true - show feedback instead of text).

Reply:
The puppose of this code is to make synchronize toolpips beetween different ButtonGroup components on the same page. I agree that it's better to lift the code. The only question - should we keep global event taht closes other ButtonGroup's tooltips?

Reply:
I don't see how the global event is useful. The tooltip of a particular button group instance must close when there is a click outside

Reply:
https://github.com/user-attachments/assets/d2d0f5c8-286c-471a-a73e-000828820f0f

Here is a demo

Reply:
I think this one is probably fine - this is consistent with how our popover works so if we decide to fix that it is better to invent a holistic solution.